x-pack/plugins/aiops/public/components/log_categorization/format_category.tsx

```diff
@@ -0,0 +1,168 @@
+ * 2.0.
+ */
+
+import React, { FC, useMemo } from 'react';
+import { EuiText, EuiHorizontalRule } from '@elastic/eui';
+import { css, SerializedStyles } from '@emotion/react';
+import type { Category } from '../../../common/api/log_categorization/types';
+import { useIsDarkTheme } from '../../hooks/use_eui_theme';
+
+interface Props {
+  category: Category;
+  count?: number;
+}
+
+// copies of the syntax colors used in EuiCode
+const tokenStyleLight = css`
+  color: #765b96;
+`;
+const wildcardStyleLight = css`
+  color: #357160;
+`;
+const tokenStyleDark = css`
+  color: #a68ac5;
+`;
+const wildcardStyleDark = css`
+  color: #54b399;
+`;
+
+interface Styles {
+  tokenStyle: SerializedStyles;
+  wildcardStyle: SerializedStyles;
+}
+
+const useStyles = (): Styles => {
+  const isDarkTheme = useIsDarkTheme();
+  return isDarkTheme
+    ? {
+        tokenStyle: tokenStyleDark,
+        wildcardStyle: wildcardStyleDark,
+      }
+    : {
+        tokenStyle: tokenStyleLight,
+        wildcardStyle: wildcardStyleLight,
+      };
+};
+
+const useCreateFormattedExample = () => {
+  const { tokenStyle, wildcardStyle } = useStyles();
+
+  function createFormattedExample(key: string, example: string): JSX.Element[] {
+    const keyTokens = key.split(' ');
+    let tempExample = ` ${example} `;
+    const positions = keyTokens.map((t) => ({
+      id: t,
+      start: 0,
+      end: 0,
+    }));
+    let offset = 0;
+    // match each token in order and record the start and end position
+    for (let i = 0; i < keyTokens.length; i++) {
+      const token = keyTokens[i];
+      const tokenReg = new RegExp(`(\\W)(${token})(\\W)`);
+
+      let j = 0;
+      const result = tokenReg.exec(tempExample);
+      if (!result) {
+        continue;
+      }
+      j = result.index;
+      const localEndOFToken = j + token.length + 1;
+      positions[i].start = offset + j + 1;
+      positions[i].end = offset + localEndOFToken;
+      // slice the example string to remove the token and preceding text
+      // to ensure we don't match future tokens in earlier text
+      tempExample = tempExample.slice(localEndOFToken);
+      offset += localEndOFToken;
+    }
+
+    tempExample = ` ${example} `;
+
+    // build up the list ot elements by chopping up the example string
+    // using the token positions found above
+    const elements: JSX.Element[] = [];
+    let pos = 0;
+    for (let i = 0; i < positions.length; i++) {
+      elements.push(
+        <span css={wildcardStyle}>{tempExample.substring(pos, positions[i].start)}</span>
+      );
+
+      elements.push(
+        <span css={tokenStyle}>{tempExample.substring(positions[i].start, positions[i].end)}</span>
+      );
+      pos = positions[i].end;
+    }
+
+    elements.push(
+      <span css={wildcardStyle}>{tempExample.substring(positions[positions.length - 1].end)}</span>
+    );
+
+    return elements;
+  }
+
+  return { createFormattedExample };
```

Comment:
Nit: This could be just `return createFormattedExample;`.

Reply:
Updated in https://github.com/elastic/kibana/pull/175320/commits/5165088307dc1ab4ceb2c93bae77662c0a65cb00